### PR TITLE
feat: Cross-account in iam-assumable-role-with-oidc

### PR DESCRIPTION
**Pull request by cebidhem**
## Description
This adds support for cross-account assumable role with OIDC

## Motivation and Context
We have a use-case where a EKS service account in an account should be able to assume a role in a EKS cluster in another account.
Fixes https://github.com/terraform-aws-modules/terraform-aws-iam/issues/157

## Breaking Changes
It doesn't bring a breaking change.

## How Has This Been Tested?
- [x] I have tested and validated these changes using one or more of the provided `examples/*` projects

In my environment, the module `iam-assumable-role-with-oidc` is used as a submodule in our custom irsa terraform module. 
I created a new irsa module, calling my updated `terraform-aws-iam//modules/iam-assumable-role-with-oidc`.
As a variable, I've set `oidc_fully_qualified_audience` to `sts.amazonaws.com` and then ran `tf apply`

Below the relevant part of it:
```  
# module.k8s_crossaccount_irsa.module.iam_assumable_role_with_oidc.aws_iam_role.this[0] will be created
  + resource "aws_iam_role" "this" {
      + arn                   = (known after apply)
      + assume_role_policy    = jsonencode(
            {
              + Statement = [
                  + {
                      + Action    = "sts:AssumeRoleWithWebIdentity"
                      + Condition = {
                          + StringLike = {
                              + oidc.eks.eu-west-1.amazonaws.com/id/my_oidc_provider_id:aud = [
                                  + "sts.amazonaws.com",
                                ]
                            }
                        }
                      + Effect    = "Allow"
                      + Principal = {
                          + Federated = "arn:aws:iam::012345678901:oidc-provider/oidc.eks.eu-west-1.amazonaws.com/id/my_oidc_provider_id"
                        }
                      + Sid       = ""
                    },
                ]
              + Version   = "2012-10-17"
            }
        )
      + create_date           = (known after apply)
      + force_detach_policies = false
      + id                    = (known after apply)
      + managed_policy_arns   = (known after apply)
      + max_session_duration  = 3600
      + name                  = "cluster-a-test-role"
      + path                  = "/"
      + tags                  = {
          + "Role" = "role-with-oidc"
        }
      + unique_id             = (known after apply)

      + inline_policy {
          + name   = (known after apply)
          + policy = (known after apply)
        }
    }

  # module.k8s_crossaccount_irsa.module.iam_assumable_role_with_oidc.aws_iam_role_policy_attachment.custom[0] will be created
  + resource "aws_iam_role_policy_attachment" "custom" {
      + id         = (known after apply)
      + policy_arn = (known after apply)
      + role       = "cluster-a-test-role"
    }
```

terraform version: `v0.14.6` 
aws provider version: `v3.37.0`

### CODE REVIEW

**Comment by cebidhem** [comment]
Hi @antonbabenko,

I saw the pre-commit hook issue, so I've run it locally but it changes almost all - if not all - the `README.md` files of the project. I'm pretty sure these changes are unwanted, but maybe I'm doing it wrong. 
Could you please point me at the right way of passing this job ?

**Comment by dgoradia** [comment]
Can this be merged, would be useful for my use-case at least.

**Comment by xometry-johnny** [comment]
same. We have a need for this as well

File: modules/iam-assumable-role-with-oidc/variables.tf [hunk 1]
```diff
@@ -89,6 +89,12 @@ variable "oidc_subjects_with_wildcards" {
   default     = []
 }
 
+variable "oidc_fully_qualified_audience" {
```
Comment (antonbabenko) [inline modules/iam-assumable-role-with-oidc/variables.tf:92]:
```suggestion
variable "oidc_fully_qualified_audiences" {
```

File: modules/iam-assumable-role-with-oidc/main.tf [hunk 2]
```diff
@@ -48,6 +48,16 @@ data "aws_iam_policy_document" "assume_role_with_oidc" {
...
         }
       }
+
+      dynamic "condition" {
+        for_each = length(var.oidc_fully_qualified_audience) > 0 ? local.urls : []
+
+        content {
+          test     = "StringLike"
+          variable = "${statement.value}:aud"
+          values   = var.oidc_fully_qualified_audience
```
Comment (antonbabenko) [inline modules/iam-assumable-role-with-oidc/main.tf:58]:
```suggestion
          values   = var.oidc_fully_qualified_audiences
```

File: modules/iam-assumable-role-with-oidc/main.tf [hunk 3]
```diff
@@ -48,6 +48,16 @@ data "aws_iam_policy_document" "assume_role_with_oidc" {
           values   = var.oidc_subjects_with_wildcards
         }
       }
+
+      dynamic "condition" {
+        for_each = length(var.oidc_fully_qualified_audience) > 0 ? local.urls : []
```
Comment (antonbabenko) [inline modules/iam-assumable-role-with-oidc/main.tf:53]:
```suggestion
        for_each = length(var.oidc_fully_qualified_audiences) > 0 ? local.urls : []
```

**Review by antonbabenko — APPROVED** [review]

**Comment by antonbabenko** [comment]
Here we go! Thank you, @cebidhem for the addition!

[v4.3.0](https://github.com/terraform-aws-modules/terraform-aws-iam/releases/v4.3.0) has been just released.

**Pull request merged**